src/enum.ts

```diff
@@ -0,0 +1,73 @@
+import { basicValidation, Check, Coerce, Convert, createAsCheck, createIsCheck, createMaybeAsCheck, createMaybeCheck, Validate } from './common';
+import { Issue, ValueProcessor } from './types';
+
+type EnumValue = string | number;
+type EnumType = { [key: string]: EnumValue };
+
+interface ValidationOptions { }
+
+class Generic<T> {
+  private keyToValue: { [key: string]: T } = {};
+  private valueToKey: { [key: string]: string } = {};
+
+  constructor(type: EnumType) {
+    for (const key in type) {
+      if (key) {
+        if (!/^\d+$/gi.test(key)) {
+          this.keyToValue[key] = type[key] as unknown as T;
+          this.valueToKey[type[key]] = key;
+        }
+      }
+    }
+  }
```

Comment:
early return is preferred and you should name even simple regexps to increase readability

```suggestion
const IS_NUMBER = /^\d+$/gi;

class Generic<T> {
  private keyToValue: { [key: string]: T } = {};
  private valueToKey: { [key: string]: string } = {};

  constructor(type: EnumType) {
    for (const key in type) {
      if (!key || IS_NUMBER.test(key)) continue;

      this.keyToValue[key] = type[key] as unknown as T;
      this.valueToKey[type[key]] = key;
    }
  }
```

Reply:
Good suggestion! I went with a private method for increased readbility. But I have a habit of not assigning Regexes to a local `const`. I hope that's ok.

Reply:
The other part of Jame's suggestion was around inverting the if statements to avoid deep nesting.

Reply:
Ah yes, I missed that suggestion. This has been done now